src/main/java/lotto/domain/LottoNumber.java

```diff
@@ -0,0 +1,89 @@
+package lotto.domain;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class LottoNumber implements Comparable<LottoNumber> {
+    private static final int BEGIN_OF_LOTTO_NUMBER = 1;
+    private static final int END_OF_LOTTO_NUMBER = 45;
+    private static final int ZERO = 0;
+    private static final Map<Integer, LottoNumber> lottoNumberRepository;
+    private final int lottoNumber;
+
+    static {
+        lottoNumberRepository = IntStream.rangeClosed(BEGIN_OF_LOTTO_NUMBER, END_OF_LOTTO_NUMBER)
+                .boxed()
+                .collect(Collectors.toMap(Function.identity(), LottoNumber::new));
+    }
+
+    public LottoNumber(int lottoNumber) {
+        this.lottoNumber = lottoNumber;
+    }
+
+    public static LottoNumber provideLottoNumber(int requestNumber) {
+
+        LottoNumber lottoNumber = lottoNumberRepository.get(requestNumber);
+
+        if (!isCorrectRange(lottoNumber)) {
```

Comment:
range 체크를 하지 않고 map 에 없다면 1~45가 아니므로 예외를 발생시키면 되지 않을까요? 🤔

```suggestion
if (lottoNumber == null) {
```